### PR TITLE
stm32l4: Timer rewrite - don't mask/unmask interrupts

### PR DESCRIPTION
**Pull request by agkaminski**
JIRA: RTOS-171

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/armv7m/stm32/l4/timer.c [hunk 1]
```diff
@@ -127,20 +136,25 @@ void timer_jiffiesAdd(time_t t)
...
 void timer_setAlarm(time_t us)
 {
-	time_t ticks = hal_timerUs2Cyc(us);
 	u32 setval;
 	spinlock_ctx_t sc;
+	time_t ticks = hal_timerUs2Cyc(us);
 
-	if (ticks > 0xffffu)
-		ticks = 0xffffu;
+	if (ticks > 0xffff0000U)
```
Comment (anglov) [inline hal/armv7m/stm32/l4/timer.c:143]: braces should be used even on single line if

File: hal/armv7m/stm32/l4/timer.c [hunk 2]
```diff
@@ -127,20 +136,25 @@ void timer_jiffiesAdd(time_t t)
...
-	setval = timer_getCnt() + (u32)ticks;
-	/* ARR has to be strictly greater than CMP */
-	if (setval == 0xffffu)
-		setval = 0;
-	*(timer_common.lptim + lptim_cmp) = setval & 0xffffu;
-	*(timer_common.lptim + lptim_ier) = 3;
+	setval = (timer_getCnt() + (u32)ticks) & 0xffff;
+
+	/* Can't have cmp == arr */
+	if (setval > 0xfffeu)
```
Comment (anglov) [inline hal/armv7m/stm32/l4/timer.c:150]: braces should be used even on single line if

**Review by anglov — APPROVED** [review]
LGTM

File: hal/armv7m/stm32/l4/timer.c (hunk 1 above)
Comment (agkaminski) [reply hal/armv7m/stm32/l4/timer.c]: Thx

**Review by anglov — APPROVED** [review]

**Pull request merged**